packages/eslint-plugin/src/rules/action-rule.ts

```diff
@@ -0,0 +1,79 @@
+import { Rule } from "eslint";
+import { CallExpression, Identifier, Literal, VariableDeclarator, ArrowFunctionExpression, Node } from "estree";
+import { isIdentifier, isLiteral } from "../lib";
+
+type ActionCallExpression =
+    CallExpression
+    & { callee: Identifier, arguments: [] | [Literal] | [ArrowFunctionExpression] | [ArrowFunctionExpression, Literal] }
+type ActionVariableDeclarator = VariableDeclarator & { id: Identifier, init: ActionCallExpression }
+
+export const actionRule: Rule.RuleModule = {
+    meta: {
+        type: 'suggestion',
+        docs: {
+            description: "Add name for every action call"
+        },
+        fixable: 'code'
+    },
```

Comment:
I'd define all possible messages here to make a code cleaner:

```suggestion
    meta: {
        type: 'suggestion',
        docs: {
            description: "Add name for every action call"
        },
        messages: {
          noname: /* message */,
          invalidName: /* message */
        },
        fixable: 'code'
    },
```

What do you think?

Reply:
Absolutely)

Reply:
why not easy? with a structure above you can pass variables like this:

```javascript
    context.report({
                messageId: "noname",
                data: { varName: varName },
                ...
              });
```

Reply:
Is there any advantages on such a solution in comparison with simple function?

`message: noname(d.id.name),`

Reply:
1. we use builtin tools
2. all rule configuration in one place

Reply:
anyway it's not typed and a bit harder to read)) (for me)